services/worker/src/worker/job_runners/config/is_valid.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import logging
+from http import HTTPStatus
+from typing import Tuple
+
+from libcommon.constants import PROCESSING_STEP_CONFIG_IS_VALID_VERSION
+from libcommon.exceptions import PreviousStepFormatError
+from libcommon.simple_cache import (
+    CacheEntryDoesNotExistError,
+    get_previous_step_or_raise,
+    get_response,
+)
+
+from worker.dtos import IsValidResponse, JobResult
+from worker.job_runners.config.config_job_runner import ConfigJobRunner
+
+
+def compute_is_valid_response(dataset: str, config: str) -> Tuple[IsValidResponse, float]:
+    """
+    Get the response of /is-valid for one specific dataset config on huggingface.co.
+
+    A dataset config is valid if any of the artifacts for any of the
+    steps is valid.
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        config (`str`):
+            A configuration name.
+    Returns:
+        `Tuple[IsValidResponse, float]`: The response (viewer, preview, search) and the progress.
+    """
+    logging.info(f"get is-valid response for {dataset=} {config=}")
+
+    split_names_response = get_previous_step_or_raise(
+        kinds=["config-split-names-from-streaming", "config-split-names-from-info"],
+        dataset=dataset,
+        config=config,
+    )
+    content = split_names_response.response["content"]
+    if "splits" not in content:
+        raise PreviousStepFormatError("Previous step did not return the expected content: 'splits'.")
+
+    preview_count = 0
+    viewer_count = 0
+    search_count = 0
+    try:
+        total = 0
+        pending = 0
+        for split_item in content["splits"]:
+            split = split_item["split"]
+            total += 1
+            try:
+                response = get_response(kind="split-is-valid", dataset=dataset, config=config, split=split)
+            except CacheEntryDoesNotExistError:
+                logging.debug("No response found in previous step for this dataset: 'split-is-valid'.")
+                pending += 1
+                continue
+            if response["http_status"] != HTTPStatus.OK:
+                logging.debug(f"Previous step gave an error: {response['http_status']}.")
+                continue
+            split_is_valid_content = response["content"]
+            preview_count += 1 if split_is_valid_content["preview"] else 0
+            viewer_count += 1 if split_is_valid_content["viewer"] else 0
+            search_count += 1 if split_is_valid_content["search"] else 0
+    except Exception as e:
+        raise PreviousStepFormatError("Previous step did not return the expected content.", e) from e
+
+    progress = (total - pending) / total if total else 1.0
+    preview = preview_count == total
```

Comment:
it means it is valid if all the splits are valid? I think it's not what we had before.

BTW, it might be clearer to do than to compare numbers

```
preview = preview or split_is_valid_content["preview"]
```

Reply:
Makes more sense, done in https://github.com/huggingface/datasets-server/pull/1618/commits/6f0bda0f19285519098a66d5eb90b82cc545f572